src/ck-core/ckfutures.h

```diff
@@ -99,7 +107,38 @@ namespace ck {
       CkReleaseFuture(handle_);
     }
     void pup(PUP::er &p) { p | handle_; }
+
+    inline bool operator==(const future<T>& other) const {
+      return (this->handle_.id == other.handle_.id) &&
+             (this->handle_.pe == other.handle_.pe);
+    }
   };
+
+  // returns a pair with the value and fulfilled future
+  template<typename InputIter, typename T = typename InputIter::value_type::value_type>
+  std::pair<T, InputIter> wait_any(InputIter first, InputIter last) {
+    std::vector<CkFutureID> handles(last - first);
```

Comment:
The construction and filling of handles should be uniform here and in `wait_all`, both should reserve their space and use the same arguments to `transform`.